nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/controller/StandardProcessorNode.java

```diff
@@ -505,24 +510,30 @@ public String getSchedulingPeriod() {
 
     @Override
     @SuppressWarnings("deprecation")
-    public synchronized void setScheduldingPeriod(final String schedulingPeriod) {
+    public synchronized void setSchedulingPeriod(final String schedulingPeriod) {
         if (isRunning()) {
             throw new IllegalStateException("Cannot modify Processor configuration while the Processor is running");
         }
 
+        final String evaluatedSchedulingPeriod = evaluateSchedulingPeriodFromParameter(schedulingPeriod);
+        final ParameterParser parameterParser = new ExpressionLanguageAgnosticParameterParser();
+        final ParameterTokenList parameterTokenList = parameterParser.parseTokens(schedulingPeriod);
+
+        parameterReferences = new ArrayList<>(parameterTokenList.toReferenceList());
+
         switch (schedulingStrategy) {
         case CRON_DRIVEN: {
             try {
-                new CronExpression(schedulingPeriod);
+                new CronExpression(evaluatedSchedulingPeriod);
             } catch (final Exception e) {
                 throw new IllegalArgumentException(
-                        "Scheduling Period is not a valid cron expression: " + schedulingPeriod);
+                        "Scheduling Period is not a valid cron expression: " + evaluatedSchedulingPeriod);
             }
         }
             break;
         case PRIMARY_NODE_ONLY:
         case TIMER_DRIVEN: {
-            final long schedulingNanos = FormatUtils.getTimeDuration(requireNonNull(schedulingPeriod),
+            final long schedulingNanos = FormatUtils.getTimeDuration(requireNonNull(evaluatedSchedulingPeriod),
                     TimeUnit.NANOSECONDS);
             if (schedulingNanos < 0) {
                 throw new IllegalArgumentException("Scheduling Period must be positive");
```

Comment:
Because these values can change outside of this method now, by changing a parameter, we need to perform these checks during validation. Because we'll be checking it there, and because we shouldn't throw an Exception if referencing a parameter that hasn't yet been set, the validation should now be removed from this method.